### DISCUSSION TITLE
weird interaction between terminal and applescript globals

### DISCUSSION THREAD

**Discussion by cfritze64** [johnlindquist/kit#307] [Errors]
The following code never displays the alert dialog unless I comment out line 1:

terminal(`echo "hello"`);
await applescript(`display alert "hello"`);

I had some scripts break (I think with beta18, although it coulda been the one of two before then). After calling terminal, execution disappeared into the ether - subsequent code not executed, no errors in the log... So I wrote this simple script to test.

**Comment by cfritze64** [comment]
More:


await applescript(`display alert "hello"`);
terminal(`echo "hello"`);

doesn't ever execute its second line, unless I comment out the first

**Reply by johnlindquist** [reply]
Thanks for these! I know @jsjoeio had some new issues with a script using applescript as well (sending keystrokes). I imagine it's something to do with the applescript helper. I'll dig into it ASAP

**Reply by jsjoeio** [reply]
I'll keep an eye on this!

**Comment by johnlindquist** [comment]
A temp fix:

```js
send("HIDE_APP")
await applescript(`display alert "hello"`)
terminal(`echo "hello"`)
```

Kit has logic to end a script when the prompt "blurs"/loses focus. Imagine clicking outside the prompt and expecting your script to exit. 

The "HIDE_APP" (again just a temp fix) will tell Kit that the blur was intentional.

I'll re-visit that blur logic and get it sorted. Something like "only exit on blur if prompt was expecting input"

**Reply by cfritze64** [reply]
logical explanations are so satisfying - many thanks!
and a single line of code (the temp fix) dropped in one spot of a multi-faceted script fixed about 8 different operations. :-)

**Comment by johnlindquist** [comment]
I'm going to send a `PROMPT_BLURRED` event to the script. This uses the same open channel that `TAB_CHANGED` uses to invoke `onTab`.

For now, it will call `exit()` by default, thus ending the script. Two benefits:

1. The channel is only open when the prompt is `await`'ing input, thus fixing this bug
2. In the future, we can make an `onBlur` handler in the script in case you want to "clean up" something

I'll try to push out a build tomorrow. (It will include a *_really_* exciting feature I've been building towards for a looong time  😉)
